Week-4/4.1/index-4.1.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+class Promise {
+    constructor(handler) {
+        this.status = 'pending';
+        this.onFulfilledCallbacks = [];
+        this.onRejectedCallbacks = [];
+
+        const resolve = value => {
+            if(this.status === 'pending') {
+                this.status = 'fulfilled';
+                this.value = value;
+                this.onFulfilledCallbacks.forEach(fn => fn(value));
+            }
+        };
+
+        const reject = value => {
+            if(this.status === 'pending') {
+                this.status = 'rejected';
+                this.value = value;
+                this.onRejectedCallbacks.forEach(fn => fn(value));
+            }
+        };
+
+        try {
+            handler(resolve, reject);
+        } catch(err) {
+            reject(err);
+        }
+    }
+
+    then(onFulfilled, onRejected) {
```

Comment:
@mr-rego it looks good. But please implement catch method as well. I should be able to call or invoke .catch on the promise returned.

Only catch should handle rejections

Reply:
Good attempt. I appreciate it